### PR TITLE
Provide Default Email Timeout Value

### PR DESCRIPTION
**Pull request by beeankha**
##### SUMMARY
Addressing issue https://github.com/ansible/awx/issues/3444

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

 - API

##### AWX VERSION
- AWX version: devel

### CODE REVIEW

File: awx/main/notifications/email_backend.py [hunk 1]
```diff
@@ -22,6 +22,14 @@ class CustomEmailBackend(EmailBackend):
     recipient_parameter = "recipients"
     sender_parameter = "sender"
 
+    def __init__(self, timeout=None, **kwargs):
+        self.timeout = timeout
+        if timeout is None:
+            self.timeout == 30
+        if not (1 <= timeout <= 120):
```
Comment (ryanpetrello) [inline awx/main/notifications/email_backend.py]: Might want to confirm that it's an integer with `isinstance` before doing this integer comparison.

File: awx/api/serializers.py [hunk 2]
```diff
@@ -4618,12 +4618,15 @@ def validate(self, attrs):
...
             object_actual = None
-        for field in notification_class.init_parameters:
+        for field, params in notification_class.init_parameters.items():
             if field not in attrs['notification_configuration']:
-                missing_fields.append(field)
+                if 'default' in params:
+                    attrs['notification_configuration'][field] = params['default']
+                else:
+                    missing_fields.append(field)
                 continue
```
Comment (ryanpetrello) [inline awx/api/serializers.py]: Do we want this `continue` to be indented into the `else` block i.e.,

```python
else:
    missing_fields.append(field)
    continue
```

If there's a default, we may still want to run the other code in this `for ...:` block to verify that the default value has the correct type, etc...

Comment (beeankha) [reply awx/api/serializers.py]: Changed!

**Review by kdelee — APPROVED** [review]
Works for me

**Review by ryanpetrello — APPROVED** [review]

**Pull request merged**